tests/collections/asr/test_diar_utils.py

```diff
@@ -35,130 +45,289 @@ def check_range_values(target, source):
     return all(bool_list)
 
 
-def matrix(mat, torch=True):
-    if torch:
-        return torch.tensor(mat)
+def check_labels(target, source):
+    bool_list = []
+    for x, y in zip(target, source):
+        bool_list.append(abs(x - y) < 1e-6)
+    return all(bool_list)
+
+
+def matrix(mat, use_tensor=True, dtype=torch.long):
+    if use_tensor:
+        mat = torch.Tensor(mat).to(dtype)
     else:
-        return np.array(mat)
+        mat = np.array(mat)
+    return mat
 
 
-def generate_mock_emb(n_emb_per_spk, perturb_sigma, emb_dim):
-    """Generate a set of artificial embedding vectors from random numbers
+def generate_orthogonal_embs(total_spks, perturb_sigma, emb_dim):
+    """Generate a set of artificial orthogonal embedding vectors from random numbers
     """
-    return torch.rand(1, emb_dim).repeat(n_emb_per_spk, 1) + perturb_sigma * torch.rand(n_emb_per_spk, emb_dim)
+    gaus = torch.randn(emb_dim, emb_dim)
+    _svd = torch.linalg.svd(gaus)
+    orth = _svd[0] @ _svd[2]
+    orth_embs = orth[:total_spks]
+    # Assert orthogonality
+    assert torch.abs(getCosAffinityMatrix(orth_embs) - torch.diag(torch.ones(total_spks))).sum() < 1e-4
+    return orth_embs
 
 
-def generate_mock_data(
+def generate_toy_data(
     n_spks=2,
     spk_dur=3,
     emb_dim=192,
-    perturb_sigma=0.01,
+    perturb_sigma=0.0,
     ms_window=[1.5, 1.0, 0.5],
     ms_shift=[0.75, 0.5, 0.25],
     torch_seed=0,
 ):
-
     torch.manual_seed(torch_seed)
-    segment_lists = []
     spk_timestamps = [(spk_dur * k, spk_dur) for k in range(n_spks)]
     emb_list, seg_list = [], []
     multiscale_segment_counts = [0 for _ in range(len(ms_window))]
+    ground_truth = []
+    random_orhtogonal_embs = generate_orthogonal_embs(n_spks, perturb_sigma, emb_dim)
```

Comment:
Spelling mistake, annoying word :D 
```suggestion
    random_orthogonal_embs = generate_orthogonal_embs(n_spks, perturb_sigma, emb_dim)
```

Reply:
fixed